scio-extra/src/test/scala/com/spotify/scio/extra/sparkey/SparkeyTest.scala

```diff
@@ -58,6 +60,44 @@ object TestCache {
     TestCache[K, V](scala.util.Random.nextString(5))
 }
 
+object SparkeyJobTest {
+  def main(cmdlineArgs: Array[String]): Unit = {
+    val (sc, args) = ContextAndArgs(cmdlineArgs)
+    val si: SideInput[SparkeyReader] = sc.sparkeySideInput(args("sparkey"))
+    sc.textFile(args("input"))
+      .withSideInputs(si)
+      .flatMap {
+        case (v, ctx) =>
+          Option(ctx(si).getAsString(v)).map(vv => s"$v -> $vv")
+      }
+      .toSCollection
+      .saveAsTextFile(args("output"))
+
+    sc.run().waitUntilDone()
+  }
+}
+
+class SparkeyIOTest extends ScioIOSpec {
+  val items = List("a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n")
+
+  "SparkeyIO" should "work with input" in {
+    val uris = items.map(SparkeyUri(_, () => ???))
+    testJobTestInput(uris)(path => SparkeyIO(path)) { case (sc, path) => sc.sparkeyUri(path) }
+  }
+
+  it should "work in a job test" in {
+    val lookup = items.map(i => i -> i)
+    val expected = items.map(i => s"$i -> $i")
+    JobTest[SparkeyJobTest.type]
+      .args("--input=input.txt", "--sparkey=mySparkey", "--output=out.txt")
+      .input(SparkeyIO("mySparkey"), List(SparkeyIO.testUri(lookup)))
```

Comment:
So this is really to allow mock data as Sparkey `SideInput` in `JobTest`. It's not quite an `IO` since `sc.read()` behaves differently than other `IO`s and `write` is unimplememnted.

How does one test Sparkey side input without this right now? And I wonder if there're somewhere in between that we can make it easier without "abusing" the `IO` mechanics?

Reply:
Correct. You can't currently test them in a `JobTest` so far as I can tell.

To test now, I've rewritten my pipelines such that there is a method taking the `SideInput[SparkeyReader]` as an input, rewritten the `JobTest` to instead use `runWithContext` and then passed a side input using a mock `SparkeyUri` like you see in `SparkeyIO.testUri`.

Reply:
I like the idea, maybe we can rename a few things to make it less misleading. I'll comment in the code.

Reply:
Integrated your comments. 

Can you explain why is a "normal" SparkeyIO doesn't make sense? I don't think I have enough context

Reply:
A normal `ScioIO` is intended to support read/write in a pipeline, like text or Avro, e.g. `sc.sparkeyFile("gs://...").map(...).saveAsSparkeyFile()`. And `T` should be something like `(String, String)`.

In this case we're not supporting reading Sparkey files as `SCollection[(String, String)]` (even though it's technically doable) but "abusing" the `ScioIO` wiring for testing. That's why I'm saying it's not a typical IO.

Also we can save the `SparkeyIO` name in case we want to support reading/writing sharded Sparkey files as `SCollection[(String, String)]`.

Reply:
Couldn't you write out the `SparkeyUri`s if you wanted to? The contents of the sparkey is just a side-effect of Uris passing through the pipeline, sorta.